pkg/daemon/ceph/osd/remove.go

```diff
@@ -137,6 +137,31 @@ func removeOSD(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInf
 	if err != nil {
 		logger.Errorf("failed to remove CRUSH host %q. %v", hostName, err)
 	}
+	// call archiveCrash to silence crash warning in ceph health if any
+	archiveCrash(clusterdContext, clusterInfo, osdID)
 
 	logger.Infof("completed removal of OSD %d", osdID)
 }
+
+func archiveCrash(clusterdContext *clusterd.Context, clusterInfo *client.ClusterInfo, osdID int) {
+	// The ceph health warning should be silenced by archiving the crash
+	crash, err := client.GetCrash(clusterdContext, clusterInfo)
+	if err != nil {
+		logger.Errorf("failed to list ceph crash. %v", err)
```

Comment:
Since in its own func now, we can return immediately when there is a failure, and simplify later code.
```suggestion
		logger.Errorf("failed to list ceph crash. %v", err)
		return
```

Reply:
done